### PR TITLE
test: Rename wait until helper to wait_until_helper

### PR DESCRIPTION
**Pull request by maflcko**
This avoids confusion with the `wait_until` member functions, which should be preferred because they take the appropriate locks and scale the timeout appropriately on their own.

### CODE REVIEW

File: test/functional/test_framework/util.py [hunk 1]
```diff
@@ -225,13 +225,13 @@ def satoshi_round(amount):
...
     """Sleep until the predicate resolves to be True.
 
     Warning: Note that this method is not recommended to be used in tests as it is
-    not aware of the context of the test framework. Using `wait_until()` counterpart
-    from `BitcoinTestFramework` or `P2PInterface` class ensures an understandable
-    amount of timeout and a common shared timeout_factor. Furthermore, `wait_until()`
+    not aware of the context of the test framework. Using the `wait_until()` members
+    from `BitcoinTestFramework` or `P2PInterface` class ensures the timeout is
+    properly scaled.  Furthermore, `wait_until()`
     from `P2PInterface` class in `mininode.py` has a preset lock.
```
Comment (hebasto) [inline test/functional/test_framework/util.py:235]: fa40d0291427922e85e579def2b616d02a01b15b
See #19760:
```suggestion
    from `P2PInterface` class in `p2p.py` has a preset lock.
```

**Review by hebasto — CHANGES_REQUESTED** [review]
Approach ACK fa1850fb59e4697622fa00c640f9fc6e18ee1a8d

File: test/functional/test_framework/util.py (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/test_framework/util.py]: thx, fixed

**Comment by theStack** [comment]
Concept ACK

**Comment by laanwj** [comment]
Code review ACK fa1cd9e1ddc6918c3d600d36eadea71eebb242b6

**Review by hebasto — APPROVED** [review]
ACK fa1cd9e1ddc6918c3d600d36eadea71eebb242b6, I have reviewed the code and it looks OK, I agree it can be merged.

**Pull request merged**